service/task.go

```diff
@@ -840,67 +791,238 @@ func (uis *UIServer) taskModify(w http.ResponseWriter, r *http.Request) {
+				TaskID:    execTask.Id,
+				Execution: execTask.Execution,
+			}
+			testResults, err := apimodels.GetCedarTestResults(ctx, opts)
+			if err != nil {
+				grip.Error(message.WrapError(err, message.Fields{
+					"task_id":   execTask.Id,
+					"execution": execTask.Execution,
+					"message":   "problem getting cedar test results",
+				}))
+				success = false
+				break
+			}
 
-	ctx, cancel := context.WithCancel(r.Context())
-	defer cancel()
+			for _, tr := range testResults {
+				allResults = append(allResults, uiTestResult{
+					TestResult: convertCedarTestResult(tr),
+					TaskId:     execTaskIDs[i],
+					TaskName:   execTask.DisplayName,
+				})
+			}
+		}
 
-	displayLogs := make(chan apimodels.LogMessage)
-	go func() {
-		defer close(displayLogs)
-		for _, line := range testLog.Lines {
-			if ctx.Err() != nil {
-				return
+		if success {
+			uiTask.TestResults = append(uiTask.TestResults, allResults...)
+			return
+		}
+	} else {
+		opts := apimodels.GetCedarTestResultsOptions{
+			BaseURL:   uis.Settings.Cedar.BaseURL,
+			TaskID:    projCtx.Task.Id,
+			Execution: projCtx.Task.Execution,
+		}
+		testResults, err := apimodels.GetCedarTestResults(ctx, opts)
+		if err != nil {
+			grip.Error(message.WrapError(err, message.Fields{
+				"task_id":   projCtx.Task.Id,
+				"execution": projCtx.Task.Execution,
+				"message":   "problem getting cedar test results",
+			}))
+			success = false
+		}
+
+		if success {
+			for _, tr := range testResults {
+				uiTask.TestResults = append(uiTask.TestResults, uiTestResult{
+					TestResult: convertCedarTestResult(tr),
+				})
 			}
-			displayLogs <- apimodels.LogMessage{
-				Type:     apimodels.TaskLogPrefix,
-				Severity: apimodels.LogInfoPrefix,
-				Version:  evergreen.LogmessageCurrentVersion,
-				Message:  line,
+			if uiTask.PartOfDisplay {
+				uiTask.DisplayTaskID = projCtx.Task.DisplayTask.Id
+			}
+			return
+		}
+	}
+
+	// If cedar test results fail, fall back to db.
+	if uiTask.DisplayOnly {
+		for i, execTask := range execTasks {
+			for _, tr := range execTask.LocalTestResults {
+				uiTask.TestResults = append(uiTask.TestResults, uiTestResult{
+					TestResult: tr,
+					TaskId:     execTaskIDs[i],
+					TaskName:   execTask.DisplayName,
+				})
 			}
 		}
-	}()
-	usr := gimlet.GetUser(ctx)
-	template := "task_log.html"
-	data := logData{Data: displayLogs, User: usr}
-	if (r.FormValue("raw") == "1") || (r.Header.Get("Content-type") == "text/plain") {
-		template = "task_log_raw.html"
-		uis.renderText.Stream(w, http.StatusOK, data, "base", template)
 	} else {
-		uis.render.WriteResponse(w, http.StatusOK, data, "base", template)
+		for _, tr := range projCtx.Context.Task.LocalTestResults {
+			uiTask.TestResults = append(uiTask.TestResults, uiTestResult{TestResult: tr})
+		}
+		if uiTask.PartOfDisplay {
+			uiTask.DisplayTaskID = projCtx.Task.DisplayTask.Id
+		}
+	}
+
+}
+
+func convertCedarTestResult(result apimodels.CedarTestResult) task.TestResult {
```

Comment:
this probably belongs in either the apimodels or testresults package (whichever doesn't cause a cycle). It should also be public

Reply:
I am inclined to keep this here, it is very specific for the old UI (it was either this or I change a bunch of HTML and javascript to understand the cedar format). I think the new UI should just use `apimodels.CedarTestResult` instead of doing this conversion.

Reply:
I'm with you on just using the one in apimodels eventually. In the meantime though, this is basically converting one internal model to another isn't it? I would think that makes more sense outside the UI code